### PR TITLE
10 post comment

### PR DESCRIPTION
**Pull request by AnnaFrancesD**

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -241,4 +241,52 @@ describe("app", () => {
...
+        .then(({ body: { comment } }) => {
+          expect(comment).toBeInstanceOf(Object);
+          expect(comment).toEqual(
+            expect.objectContaining({
+              comment_id: 7,
+              author: "bainesface",
+              body: "I was the werewolf...",
+              review_id: 3,
+              votes: 0,
+            })
```
Comment (Dutton1990) [inline __tests__/endpoints.test.js:264]: you can add in created_at and just expect.any(String)

**Comment by Dutton1990** [comment]
you can add in tests for an invalid article_id and whether the article_id exists >>> 400/404

you can also alter one of your tests, if the username doesn't exist in the database PSQL should return an error itself and it would be a 404

File: models/reviews-models.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const connection = require("../db/connection");
 
 exports.fetchReviewByReviewId = (id) => {
-  if (typeof Number(id) === "number" && !isNaN(Number(id))) {
+  if (!isNaN(Number(id))) {
```
Comment (emilyb93) [inline models/reviews-models.js]: Id want you to instead let SQL handle this. SQL will throw an error if the data type is incorrect so we should let SQL decide if this is an incorrect data type as this is essentially repeating code/functionality

File: models/reviews-models.js [hunk 3]
```diff
@@ -70,6 +70,25 @@ WHERE comments.review_id = $1;`,
...
 };
 
-exports.updateReviewByReviewId = (reviewId, reviewUpdate) => {
-  console.log(reviewUpdate);
+exports.insertComment = (id, newComment) => {
+  if (
+    newComment.username &&
+    newComment.body &&
+    Object.keys(newComment).length === 2 &&
+    !isNaN(Number(id))
```
Comment (emilyb93) [inline models/reviews-models.js]: Same as what i said above, let SQL handle the data type checks

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Your error handlers will likely still pass the tests, but just remove anything that is checking data type before passing it to SQL. We get that check for free with SQL so we dont need to write the extra code.

Please post another nchelp pr after these changes

File: app.js [hunk 4]
```diff
@@ -18,9 +19,32 @@ app.patch("/api/reviews/:review_id", patchReviewByReviewId);
...
+  if (err.code === "23503") {
+    res.status(404).send({ msg: "Not Found" });
+  }
+  next(err);
+});
+
+app.use((err, req, res, next) => {
+  if (err.code === "23502") {
+    res.status(400).send({ msg: "Bad Request" });
+  }
```
Comment (Dutton1990) [inline app.js:44]: you can combine these into the same block!

**Comment by Dutton1990** [comment]
once you've made that last change feel free to merge, it all looks good to me :)

**Pull request merged**